### PR TITLE
Use E in M306 T to specify autotuned extruder

### PR DESCRIPTION
**Pull request by GMagician**
Let MPC autotune to specify extruder

It should implement #25325 but I can't test it.

### CODE REVIEW

File: Marlin/src/module/temperature.cpp [hunk 1]
```diff
@@ -914,27 +914,29 @@ volatile bool Temperature::raw_temps_ready = false;
     };
 
     struct OnExit {
+      heater_id_t e;
```
Comment (GMagician) [inline Marlin/src/module/temperature.cpp]: Not sure OnExit local variable is right approach

Comment (GMagician) [reply Marlin/src/module/temperature.cpp]: Not type has been changed to uint8_t but question is still valid

**Comment by thinkyhead** [comment]
`MPCTEMPBED` might be a thing in the future, so if you want to keep using `heater_id_t` it's probably fine.

**Comment by GMagician** [comment]
IMO when bed will be implemented it may be renamed to heater_id, but I'm ready to change it now if you think it's better

**Comment by thinkyhead** [comment]
> IMO when bed will be implemented it may be renamed to heater_id, but I'm ready to change it now if you think it's better

This can wait. I'll go ahead and merge what we have now.

**Pull request merged**